mesonbuild/compilers/cuda.py

```diff
@@ -616,8 +616,19 @@ def get_options(self) -> 'KeyedOptionDictType':
         return opts
 
     def _to_host_compiler_options(self, options: 'KeyedOptionDictType') -> 'KeyedOptionDictType':
+        """
+        Convert an NVCC Option set to a host compiler's option set.
+        """
+
+        # We must strip the -std option from the host compiler option set, as NVCC has
+        # its own -std flag that may not agree with the host compiler's.
         overrides = {name: opt.value for name, opt in options.items()}
-        return OptionOverrideProxy(overrides, self.host_compiler.get_options())
+        overrides.pop(OptionKey('std', machine=self.for_machine,
+                                       lang=self.host_compiler.language), None)
+        host_options = self.host_compiler.get_options().copy()
+        if 'std' in host_options:
+            del host_options['std']
```

Comment:
@dcbaker I need some help here with Mypy type annotations. How do I express the idea that I want to delete the `std` option both from the host compiler and from the overrides, that special flag being handled already beginning at line 638?

Reply:
@dcbaker How should I handle the above?

Reply:
Unless someone can come up with a better solution soon, disabling mypy for that line is fine.